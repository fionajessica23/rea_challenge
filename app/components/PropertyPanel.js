import React from 'react';
import PropTypes from 'prop-types';
import PropertyCard from './PropertyCard';
import styles from './PropertyPanel.scss';

const PropertyPanel = ({ headingText, results, handlePropBtnClick, buttonText }) => {
  const hasResults = results.length > 0;

  return hasResults ?
    <div>
      <h2 className={styles.subHeadingText}>{headingText}</h2>
      <ul>
        {results.map(result => (
          <li key={result.id}>
            <PropertyCard
              listing={result}
              handlePropBtnClick={handlePropBtnClick}
              buttonText={buttonText}
            />
          </li>))}
      </ul>
    </div>
    :
    <div>
      <h2 className={styles.subHeadingText}>{headingText}</h2>
    </div>;
};

PropertyPanel.propTypes = {
  headingText: PropTypes.string.isRequired,
  results: PropTypes.arrayOf(PropTypes.shape({
    price: PropTypes.string.isRequired,
    agency: PropTypes.shape({
      brandingColors: PropTypes.shape({
        primary: PropTypes.string.isRequired,
      }).isRequired,
      logo: PropTypes.string.isRequired,
    }).isRequired,
    id: PropTypes.string.isRequired,
    mainImage: PropTypes.string.isRequired,
  })),
  handlePropBtnClick: PropTypes.func.isRequired,
  buttonText: PropTypes.string.isRequired,
};

PropertyPanel.defaultProps = {
  results: [],
};

export default PropertyPanel;
