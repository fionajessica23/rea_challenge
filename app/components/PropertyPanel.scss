@import "./style-variables.scss";

.wrapper {
  width: 100%;
}

.subHeadingText {
  margin: 0;
  text-align: center;
  font-size: $xlarge-font;
  font-weight: $normal-font-weight;
}

.list {
  margin: 0;
  padding: 0;
}

.item {
  list-style: none;
}