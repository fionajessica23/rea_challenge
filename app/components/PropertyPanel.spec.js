import React from 'react';
import renderer from 'react-test-renderer';
import PropertyPanel from './PropertyPanel';

const defaultProps = {
  headingText: 'some heading text',
  results: [
    {
      price: '$650,000',
      agency: {
        brandingColors: {
          primary: '#000000',
        },
        logo: 'http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif',
      },
      id: '9',
      mainImage: 'http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg',
    },
    {
      price: '$300,000',
      agency: {
        brandingColors: {
          primary: '#111111',
        },
        logo: 'http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif',
      },
      id: '5',
      mainImage: 'http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg',
    },
  ],
  handlePropBtnClick: () => { },
  buttonText: 'some button text',
};

it('renders Property Panel with props correctly', () => {
  const tree = renderer
    .create(<PropertyPanel {...defaultProps} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it('renders Property Panel with no listing', () => {
//   const tree = renderer;
//   const propWithNoListing = { ...defaultProps, results: [] }
//     .create(<PropertyPanel {...propWithNoListing} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
