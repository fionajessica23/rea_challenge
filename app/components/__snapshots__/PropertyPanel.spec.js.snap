// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Property Panel with listings 1`] = `
<div>
  <h2
    className="mockClassName-subHeadingText"
  >
    some heading text
  </h2>
  <ul>
    <li>
      <div
        className="mockClassName-container"
      >
        <div
          className="mockClassName-header"
          style={
            Object {
              "backgroundColor": "#000000",
            }
          }
        >
          <img
            alt="Agency logo"
            src="http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif"
          />
        </div>
        <div
          className="mockClassName-image"
        >
          <img
            alt="Property 9"
            src="http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
          />
        </div>
        <div
          className="mockClassName-footer"
        >
          <div
            className="mockClassName-priceHolder"
          >
            $650,000
          </div>
          <button
            className="mockClassName-buttonAddRemove"
            onClick={[Function]}
          >
            some button text
          </button>
        </div>
      </div>
    </li>
    <li>
      <div
        className="mockClassName-container"
      >
        <div
          className="mockClassName-header"
          style={
            Object {
              "backgroundColor": "#111111",
            }
          }
        >
          <img
            alt="Agency logo"
            src="http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif"
          />
        </div>
        <div
          className="mockClassName-image"
        >
          <img
            alt="Property 5"
            src="http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
          />
        </div>
        <div
          className="mockClassName-footer"
        >
          <div
            className="mockClassName-priceHolder"
          >
            $300,000
          </div>
          <button
            className="mockClassName-buttonAddRemove"
            onClick={[Function]}
          >
            some button text
          </button>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`renders Property Panel with no listing 1`] = `
<div>
  <h2
    className="mockClassName-subHeadingText"
  >
    some heading text
  </h2>
</div>
`;
